tests/queue.spec.ts

```diff
@@ -150,6 +151,41 @@ describe('ARRAY LIKE USAGE', () => {
 	});
 });
 
+describe('Iteration', () => {
+	it('should work with empty queue', () => {
+		const iter = new QueueIterator(queue);
```

Comment:
This test covers exactly what we talked about: calling next doesn't throw, res.value is null, and res.done is true. Break the test into 3 specific tests, each testing 1 thing. In general, avoid phrases like "works as expected", "works correctly", "valid input", etc with tests.

Imagine some time in the future someone makes a change to this library and suddenly the `should work with one element` test starts failing. But it fails when running on Jenkins, not when running locally. Without inspecting the code, could they tell what might be wrong? They won't know what `working`, `valid`, or `correct` means if they don't know the original spec. That's especially true if many things break all at once and you can't tell how it should work. The solution is specifically declaring the 1 thing each unit test is testing, and the expected outcome.

- [ ] should not throw when calling iter.next on an empty queue
- [ ] should return `true` for done when queue is empty.
- [ ] should return `null` for value when queue is empty.